extensions/amp-youtube/0.1/amp-youtube.js

```diff
@@ -190,17 +195,30 @@ class AmpYoutube extends AMP.BaseElement {
 
     this.iframe_ = iframe;
 
-
-
-    this.win.addEventListener(
-        'message', event => this.handleYoutubeMessages_(event));
+    this.unlistenMessage_ = listen(
+      this.win,
+      'message',
+      this.handleYoutubeMessages_.bind(this)
+    );
 
     return this.loadPromise(iframe)
         .then(() => this.listenToFrame_())
         .then(() => this.playerReadyPromise_);
   }
 
   /** @override */
+  unlayoutCallback() {
```

Comment:
We also need to reset `#playerState`, `#playerReadyPromise`, and `#playerReadyResolver`.